### PR TITLE
feat(#20): implement TD3 (Twin Delayed DDPG)

### LINKED ISSUE #20: Baseline: Implement TD3 (Twin Delayed DDPG)
Implement TD3 in `rl_agents/td3.py` as a deterministic alternative to SAC. Include twin Q-networks, delayed policy updates, and target policy smoothing.

### PR DESCRIPTION
**Pull request by andnp**
Closes #20

## Summary
Implements the TD3 (Twin Delayed Deep Deterministic Policy Gradient) algorithm in `rl-agents`.

### Key components:
- **TD3Config**: chex dataclass with TD3-specific hyperparameters (policy delay, target noise, noise clip, exploration noise)
- **Actor**: deterministic policy with tanh-bounded output scaled to action space
- **Critic**: twin Q-networks for reducing overestimation bias
- **make_train**: JIT-compatible training loop with:
  - Delayed policy updates (every `POLICY_DELAY` steps)
  - Target policy smoothing (clipped noise on target actions)
  - Soft target network updates (Polyak averaging)
  - Configurable exploration noise

### Tests:
- 9 small tests: config defaults, network shapes, actor output bounds, soft update correctness
- 5 medium tests: make_train JIT compilation, gradient flow verification

### CODE REVIEW

File: tests/small/test_rl_agents_td3.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            mutable_cfg.LR = 0.1
+            raise AssertionError("Should have raised")
+        except AttributeError:
+            pass
+
+    def test_custom_config(self):
+        cfg = TD3Config(POLICY_DELAY=3, TAU=0.01, EXPLORATION_NOISE=0.2)
+        assert cfg.POLICY_DELAY == 3
+        assert cfg.TAU == 0.01
+        assert cfg.EXPLORATION_NOISE == 0.2
```
Comment (andnp) [inline tests/small/test_rl_agents_td3.py]: These are useless and noisy tests. Let's delete

File: libs/rl-agents/src/rl_agents/td3.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+def _critic_apply(module: Critic, variables: VariableDict, x: jax.Array, a: jax.Array) -> jax.Array:
+    return cast(jax.Array, module.apply(variables, x, a))
+
+
+def _actor_apply(module: Actor, variables: VariableDict, x: jax.Array) -> jax.Array:
+    return cast(jax.Array, module.apply(variables, x))
+
+
+def make_train(config: TD3Config, env: object, env_params: object | None = None):
```
Comment (andnp) [inline libs/rl-agents/src/rl_agents/td3.py]: Big monolith functions like this can be hard to read. I'd prefer if we could try to break it up a little and have some module-scoped functions with fewer lambda captures.
